fdbclient/SpecialKeySpace.actor.h

```diff
@@ -62,29 +62,36 @@ class SpecialKeySpace {
 		impls = KeyRangeMap<SpecialKeyRangeBaseImpl*>(nullptr, spaceEndKey);
 		range = KeyRangeRef(spaceStartKey, spaceEndKey);
 	}
-	void registerKeyRange(const KeyRangeRef& kr, SpecialKeyRangeBaseImpl* impl) {
+	void registerKeyRange(SpecialKeySpace::MODULE module, const KeyRangeRef& kr, SpecialKeyRangeBaseImpl* impl) {
 		// range check
 		// TODO: add range check not to be replaced by overlapped ones
 		ASSERT(kr.begin >= range.begin && kr.end <= range.end);
 		// make sure the registered range is not overlapping with existing ones
 		// Note: kr.end should not be the same as another range's begin, although it should work even they are the same
 		ASSERT(impls.rangeContaining(kr.begin) == impls.rangeContaining(kr.end) && impls[kr.begin] == nullptr);
 		impls.insert(kr, impl);
+		// Set module for the range
+		implToModule[impl] = module;
 	}
+	KeyRangeMap<SpecialKeyRangeBaseImpl*>& getImpls() { return impls; }
+	KeyRangeRef getKeyRange() const { return range; }
+	std::map<SpecialKeyRangeBaseImpl*, SpecialKeySpace::MODULE> getImplToModuleMap() const { return implToModule; }
```

Comment:
Return a const reference, and probably use unordered_map

Reply:
Done. Thanks.